### PR TITLE
Remove most pre-dependencies on deb packages

### PR DESCRIPTION
**Pull request by tuliom**
5 of our deb packages have a Pre-Depends relationship because their
postinst(configure) execution depend on the availability of ldconfig.
However, many other packages had a Pre-Depends relationship without
having a pre-configure dependency.

### CODE REVIEW

**Review by ThinkOpenly — APPROVED** [review]

**Comment by tuliom** [comment]
```
dh_testdir: error: depends-field appears twice in the same stanza of debian/control. First time on line 43, second time: 44
```
That needs to be fixed.

**Comment by tuliom** [comment]
I updated this PR with 2 changes:
1. Fixed (hopefully!) the previous build failures.
2. Added a new patch that removes the rest of the `Pre-Depends` entries.

I still believe this is still draft material because:
1. We need to decide if we're going to change previous versions.
2. Test. I didn't test my changes and these patches do require pedantic tests.

**Review by ThinkOpenly — APPROVED** [review]

**Comment by ThinkOpenly** [comment]
> 1. Fixed (hopefully!) the previous build failures.

:+1: 

> 2. Added a new patch that removes the rest of the `Pre-Depends` entries.

:+1:

> I believe this is still draft material because:
> 
> 1. We need to decide if we're going to change previous versions.

I would be OK _not_ changing previous versions, because these types of changes always seem to be more complicated than they appear.

> 2. Test. I didn't test my changes and these patches do require pedantic tests.

I suspect we can be no worse off than without the changes, because it is currently a failure when installing multiple packages simultaneously using `dpkg`.  I further suspect these changes will only impact that scenario, and for the better.  These changes only obviously impact _installation_, so latent bugs at package removal seem unlikely.  (But, my suspicions can't replace testing!  :-)

**Comment by rzinsly** [comment]
> > 1. We need to decide if we're going to change previous versions.
> 
> I would be OK _not_ changing previous versions, because these types of changes always seem to be more complicated than they appear.
> 

I agree.

Seems we need changes in ck_provides as well.

**Comment by tuliom** [comment]
@ThinkOpenly , @rzinsly , I modified the patches to change only AT next and fixed the errors on ck_provides.
I'm also removing the draft status of this PR. I think we can merge this after it gets enough approvals.

**Comment by tuliom** [comment]
So far, the failures on AT <= 12.0 are related to Python.
Waiting on PR #1749 in order to progress with this.

**Comment by ThinkOpenly** [comment]
> @ThinkOpenly , @rzinsly , I modified the patches to change only AT next and fixed the errors on ck_provides.
> I'm also removing the draft status of this PR. I think we can merge this after it gets enough approvals.

Still LGTM, and my approval is still in place.

**Review by rzinsly — APPROVED** [review]
LGTM

**Comment by tuliom** [comment]
I rebased the PR on top of the master branch hoping that all the tests pass this time.

**Comment by tuliom** [comment]
My last rebase did not have all the fixes on AT 10.0.  Trying again...

**Comment by tuliom** [comment]
33 successful checks and 2 approvals.
I think this is ready to be merged.
Thanks!

**Pull request merged**
